cns/cnsclient/cnsclient_test.go

```diff
@@ -242,8 +236,47 @@ func TestCNSClientRequestAndRelease(t *testing.T) {
 		t.Fatalf("Number of available IP addresses expected to be 1, actual %+v", ipaddresses)
 	}
 
-	if ipaddresses[0].IPAddress != desiredIpAddress && ipaddresses[0].State != cns.Available {
+	if ipaddresses[0].IPAddress != desiredIpAddress && ipaddresses[0].State != cns.Allocated {
 		t.Fatalf("Available IP address does not match expected, address state: %+v", ipaddresses)
 	}
 	fmt.Println(ipaddresses)
+
+	// release requested IP address, expect success
+	err = cnsClient.ReleaseIPAddress(orchestratorContext)
+	if err != nil {
+		t.Fatalf("Expected to not fail when releasing IP reservation found with context: %+v", err)
+	}
 }
+
+
+func TestCNSClientDebugAPI(t *testing.T) {
+	podName := "testpodname"
+	podNamespace := "testpodnamespace"
+	desiredIpAddress := "10.0.0.5"
+
+	secondaryIps := make([]string, 0)
+	secondaryIps = append(secondaryIps, desiredIpAddress)
+	cnsClient, _ := InitCnsClient("")
+
+	addTestStateToRestServer(t, secondaryIps)
+
+	podInfo := cns.KubernetesPodInfo{PodName: podName, PodNamespace: podNamespace}
+	orchestratorContext, err := json.Marshal(podInfo)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// request IP address
+	_, err1 := cnsClient.RequestIPAddress(orchestratorContext)
+	if err1 != nil {
+		t.Fatalf("get IP from CNS failed with %+v", err1)
+	}
+
+	//test for debug api/cmd to get inmemory data from HTTPRestService
+	inmemory, err := cnsClient.GetHTTPServiceData()
+	if err != nil {
+		t.Fatalf("Get in-memory http REST Struct failed %+v", err)
+	}
+	fmt.Println(inmemory.HttpRestServiceData)
```

Comment:
Should make some assertions on the response. What specific data do you expect in `inmemory`?

Reply:
added few checks on this function